lib/solr.rb

```diff
@@ -24,12 +24,12 @@ def configure
       yield configuration
     end
 
-    def delete_by_id(id, commit: false)
-      Solr::Delete::Request.new(id: id).run(commit: commit)
+    def delete_by_id(id, commit: false, core_name: Solr.configuration.default_core_name)
```

Comment:
I think it should be uniform everywhere, e.g. `delete_by_id(core: :listing)` and `Solr.configuration.default_core` for all user facing code.